### ISSUE TITLE
Separate all the functionality from dataconverter to a reader utilities subpackge or module.  

### ISSUE THREAD

**Issue by RubelMozumder** [FAIRmat-NFDI/pynxtools#258]
### What happened?

Move all the functionalities that are not directly involved to the dataconverter but rather used in readers e.g. `extract_atom_types` should move to the reader utils

What do you think here @domna, @lukaspie, @sherjeelshabih, @mkuehbach?

**Comment by domna** [comment]
I think we mostly keep it in the `utils.py` file already. But it's a good point to actually clean this up and offer this as a utils package also if people want to use it from their reader plugins.

**Comment by RubelMozumder** [comment]
> I think we mostly keep it in the `utils.py` file already. But it's a good point to actually clean this up and offer this as a utils package also if people want to use it from their reader plugins.

you are suggesting having another `python` package only for reader `utils`?

**Comment by domna** [comment]
> > I think we mostly keep it in the `utils.py` file already. But it's a good point to actually clean this up and offer this as a utils package also if people want to use it from their reader plugins.
> 
> you are suggesting having another `python` package only for reader `utils`?

No, just as a sub-package to pynxtools, e.g., `pynxtools.utils` or so. Currently, it's in `pynxtools.dataconverter.utils`, which is probably also fine.
